modules/apps/address/address-impl/src/main/java/com/liferay/address/internal/util/CompanyCountriesUtil.java

```diff
@@ -143,50 +155,74 @@ public static void processCountryRegions(
 				_log.debug("Regions found for country " + a2);
 			}
 
-			for (int i = 0; i < regionsJSONArray.length(); i++) {
-				try {
+			if (regionsJSONArray.length() == 0) {
+				return;
+			}
+
+			CTPersistence<Region> ctPersistence =
+				regionLocalService.getCTPersistence();
+
+			DataSource dataSource = ctPersistence.getDataSource();
+
+			try (Connection connection = dataSource.getConnection();
+				PreparedStatement regionPreparedStatement =
+					AutoBatchPreparedStatementUtil.autoBatch(
+						connection,
+						StringBundler.concat(
+							"INSERT INTO Region (mvccVersion, ctCollectionId",
+							", uuid_, regionId, companyId, userId, createDate",
+							", modifiedDate, countryId, active_, name",
+							", \"position\", regionCode) VALUES (0, 0, ?, ?, ",
```

Comment:
@smottal the performance goes back to normal, because you have a broken SQL syntax here, and the insert did not happen at all. The position field can not be quoted. This prepared statement creation fails with `Caused by: org.hsqldb.HsqlException: user lacks privilege or object not found: position` And this try catch block is only doing debug logging that why @111Anna did not catch it. Fatal error can not be logged as debug, it must be error level.

Please fix it and go through CI first for your next pull.

Reply:
Hi @shuyangzhou, I'm really sorry about this, I didn't realize it because in postgres everything also works with the escaped column.

I updated the PR https://github.com/liferay-core-infra/liferay-portal/pull/6529 with the requested changes, including the one related to the catch of the exception. I also launched the CI [here](https://github.com/smottal/liferay-portal/pull/12). 

If you can take a look at it. Thank you